Sources/StreamChat/Database/DTOs/UserDTO.swift

```diff
@@ -30,10 +33,24 @@ class UserDTO: NSManagedObject {
         request.predicate = NSPredicate(format: "id == %@", userId)
         return request
     }
-    
+
     override func willSave() {
         super.willSave()
-        
+
+        // When user changed, we need to propagate this change to members and current user
+        if hasPersistentChangedValues {
+            if !(currentUser?.hasChanges ?? false) {
+                // this will not change object, but mark it as dirty, triggering updates
+                currentUser?.unreadChannelsCount = currentUser?.unreadChannelsCount ?? 0
+            }
```

Comment:
wouldn't this be more readable and simple using `if let currentUser = currentUser, currentUSer.hasChanges { }` ?

Reply:
WAIT, we shouldn't check `currentUser`'s changes here, right?